builder/comp-builder.nix

```diff
@@ -22,16 +22,17 @@
 
 , doCheck ? component.doCheck || haskellLib.isTest componentId
 , doCrossCheck ? component.doCrossCheck || false
-, dontPatchELF ? true
-, dontStrip ? true
+, dontPatchELF ? component.dontPatchELF
+, dontStrip ? component.dontStrip
 
-, static ? stdenv.hostPlatform.isMusl
-, deadCodeElimination ? true
+, enableStatic ? component.enableStatic
+, enableShared ? component.enableShared
```

Comment:
I had to add this because I was trying to build a docker image. The problem was that one of our binaries uses `ekg`, so the path to the `ekg` data directory appeared in my binary. This causes nix to pull in the whole `ekg` package, with the shared libraries. Because the shared libraries have a link to ghc, this caused the ghc nix path to get included. Which then caused the gcc nix path to get included. Suddenly, my docker image was nearly 2GB instead of 100MB. Disabling shared libraries from being built fixes this problem for me and is probably useful to other people for other reasons.

Reply:
Great -- I was wondering why gcc was in the closure of everything.

Why do the shared libraries have a link to ghc? Do we need to put the RTS in a separate output or something?

In any case there is probably not a great demand from users for dynamically linking Haskell libraries.

Reply:
@rvl that's a neat idea! We might be able to construct something like that, by going down the path with reinstallable lib:ghc a bit further. I'd venture a geuss that this applies to all shipped libraries; hence maybe we should split ghc and the shipped package db into two derivations.

For now I agree that dynamically linked haskell libs (outside of an existing nix installation) is a bit out of scope.

Reply:
See #347

Reply:
Just in case somebody else stumbles upon this, for the ekg package the following snippet did the trick for me:

```
modules = [
  ({ pkgs, config, ...}: {
      packages = {
        ekg.enableSeparateDataOutput = true;
      };
  }) ];
```